### PR TITLE
ModuleService and repository with tests

### PR DESCRIPTION
**Pull request by levineuensch**
Added in the implementation for ModuleService with tests, added ModuleRepository

### CODE REVIEW

File: src/test/java/com/revature/RevAssure/service/ModuleServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Test
+    void saveNewModuleTest(){
+        when(repo.save(module)).thenReturn(module);
+        assertEquals(1,service.saveNewModule(module).getId());
+    }
+
+    @Test
+    void findAllModulesTest(){
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/service/ModuleServiceTest.java:60]: you should test that the list is either the same as your expected list, or that its contents are the same as your expected list. Two lists can have the same size but different contents

File: src/test/java/com/revature/RevAssure/service/ModuleServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Test
+    void updateExistingModuleTest(){
+        module.setName("newTestName");
+        when(repo.save(module)).thenReturn(module);
+        assertEquals("newTestName",service.saveExistingModule(module).getName());
+    }
+
+    @Test
+    void deleteExistingModuleTest(){
```
Comment (cptvictor) [inline src/test/java/com/revature/RevAssure/service/ModuleServiceTest.java:73]: delete shouldn't return anything, verify delete line is run (reference deleteTopicTest() in TopicServiceTest)

**Review by cptvictor — CHANGES_REQUESTED** [review]
please make changes to your tests

**Review by cptvictor — APPROVED** [review]
lgtm

**Review by AlwynZ — APPROVED** [review]
lgtm

**Pull request merged**